### PR TITLE
Dbt cloud enterprise

### PR DESCRIPTION
**Pull request by radbrt**
## Summary
Add argument for custom domain to dbt Cloud task, letting Enterprise customers use the task. Enterprise customers have a custom subdomain like "cloud.corp.getdbt.com".

## Changes
Add a single argument, "domain", to the dbt.DbtCloudRunJob class init and run methods. The argument is optional and defaults to "cloud.getdbt.com".

## Importance
This will let enterprise customers of dbt Cloud use the dbt Cloud task.

## Checklist

This PR:

- [Y] adds new tests (if appropriate)
- [Y] adds a change file in the `changes/` directory (if appropriate)
- [Y] updates docstrings for any new functions or function arguments, including `docs/outline.toml` for API reference docs (if appropriate)

### CODE REVIEW

File: src/prefect/tasks/dbt/dbt.py [hunk 1]
```diff
@@ -262,6 +263,7 @@ def __init__(
         token_env_var_name: str = "DBT_CLOUD_TOKEN",
         wait_for_job_run_completion: bool = False,
         max_wait_time: int = None,
+        domain: str = None,
```
Comment (desertaxle) [inline src/prefect/tasks/dbt/dbt.py:266]: I recommend setting the default value of this parameter to `cloud.getdbt.com` like so:
```suggestion
        domain: str = "cloud.getdbt.com",
```

File: src/prefect/tasks/dbt/dbt.py [hunk 2]
```diff
@@ -231,6 +231,7 @@ class DbtCloudRunJob(Task):
         - max_wait_time (int, optional): The number of seconds to wait for the dbt Cloud
             job to finish.
             Used only if wait_for_job_run_completion = True.
+        - domain (str, optional): Custom domain for API call.
```
Comment (desertaxle) [inline src/prefect/tasks/dbt/dbt.py:234]:
```suggestion
        - domain (str, optional): Custom domain for API call. Defaults to `cloud.getdbt.com`.
```

File: src/prefect/tasks/dbt/dbt.py [hunk 3]
```diff
@@ -329,6 +334,7 @@ def run(
             - max_wait_time (int, optional): The number of seconds to wait for the dbt Cloud
                 job to finish.
                 Used only if wait_for_job_run_completion = True.
+            - domain (str, optional): Custom domain for API call.
```
Comment (desertaxle) [inline src/prefect/tasks/dbt/dbt.py:337]:
```suggestion
        - domain (str, optional): Custom domain for API call. Defaults to `cloud.getdbt.com`.
```

File: src/prefect/tasks/dbt/dbt_cloud_utils.py [hunk 4]
```diff
@@ -117,7 +127,8 @@ def wait_for_job_run(
         - account_id (int): dbt Cloud account ID
         - token (str): dbt Cloud token
         - run_id (int): dbt Cloud job run ID
-        - max_wait_time: the number od seconds to wait for the job to complete
+        - max_wait_time (int): the number od seconds to wait for the job to complete
+        - domain (str): The domain the function should call, normally cloud.getdbt.com
```
Comment (desertaxle) [inline src/prefect/tasks/dbt/dbt_cloud_utils.py:131]:
```suggestion
        - domain (str): The domain the function should call (e.g. `cloud.getdbt.com`).
```

File: src/prefect/tasks/dbt/dbt_cloud_utils.py [hunk 5]
```diff
@@ -166,6 +180,7 @@ def list_run_artifact_links(
         - account_id (int): dbt Cloud account ID
         - run_id (int): dbt Cloud job run ID
         - token (str): dbt Cloud token
+        - domain (str): The domain the function should call, normally cloud.getdbt.com
```
Comment (desertaxle) [inline src/prefect/tasks/dbt/dbt_cloud_utils.py:183]:
```suggestion
        - domain (str): The domain the function should call (e.g. `cloud.getdbt.com`).
```

File: src/prefect/tasks/dbt/dbt_cloud_utils.py [hunk 6]
```diff
@@ -82,6 +87,7 @@ def trigger_job_run(
         - job_id (int): dbt Cloud job ID
         - token (str): dbt Cloud token
         - cause (str): the reason describing why the job run is being triggered
+        - domain (str): the domain the function should call, default cloud.getdbt.com
```
Comment (desertaxle) [inline src/prefect/tasks/dbt/dbt_cloud_utils.py:90]:
```suggestion
        - domain (str): The domain the function should call (e.g. `cloud.getdbt.com`).
```

**Review by desertaxle — CHANGES_REQUESTED** [review]
Thanks for your contribution! This looking good! I requested a few small tweaks. After those are addressed this will be good to go!

**Review by desertaxle — APPROVED** [review]

**Comment by madkinsz** [comment]
Thanks for contributing!

**Pull request merged**